grpc-server-runtime/src/test/groovy/io/micronaut/grpc/server/health/GrpcServerHealthSpec.groovy

```diff
@@ -0,0 +1,127 @@
+package io.micronaut.grpc.server.health
+
+import io.grpc.Channel
+import io.grpc.Status
+import io.grpc.StatusRuntimeException
+import io.grpc.health.v1.HealthCheckRequest
+import io.grpc.health.v1.HealthCheckResponse
+import io.grpc.health.v1.HealthGrpc
+import io.grpc.protobuf.services.HealthStatusManager
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Factory
+import io.micronaut.context.annotation.Property
+import io.micronaut.context.annotation.Requires
+import io.micronaut.core.io.socket.SocketUtils
+import io.micronaut.grpc.annotation.GrpcChannel
+import io.micronaut.grpc.server.GrpcEmbeddedServer
+import io.micronaut.grpc.server.GrpcServerChannel
+import jakarta.inject.Inject
+import jakarta.inject.Singleton
+import spock.lang.Specification
+
+@Property(name = "spec.name", value = "GrpcServerHealthSpec")
+class GrpcServerHealthSpec extends Specification {
+
+    void "grpc HealthCheck is enabled by default, and reports SERVING"() {
+        given:
+        GrpcEmbeddedServer server = createServer()
+
+        when:
+        def bean = server.applicationContext.getBean(TestBean)
+        def result = bean.check()
+
+        then:
+        result.status == HealthCheckResponse.ServingStatus.SERVING
+
+        cleanup:
+        server.close()
+    }
+
+    void "disabling the health check results in an UNIMPLEMENTED exception"() {
+        given:
+        GrpcEmbeddedServer server = createServer('grpc.server.health.grpc.enabled': 'false')
```

Comment:
You can use a `StringUtils.FALSE` if you want.

Reply:
That's a lot more typing 😉